src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java

```diff
@@ -436,13 +440,16 @@ public void appendEnd(final StringBuffer buffer, final Object object) {
      * @param buffer  the <code>StringBuffer</code> to populate
      * @since 2.0
      */
+    @SuppressWarnings("index") /* #1 - i < sepLen.length() and len >= sepLen => i < len.length()
```

Comment:
The root cause of this warning seems to be that `len >= sepLen` is the comparison that makes this safe, and the SameLen checker cannot handle non-equal array sizes. Can you update the comment to mention that?